mayastor/src/lvs/lvol.rs

```diff
@@ -330,4 +353,59 @@ impl Lvol {
             }
         }
     }
+
+    /// Format snapshot name
+    /// base_name is the nexus or replica UUID
+    pub fn format_snapshot_name(base_name: &str, snapshot_time: u64) -> String {
+        format!("{}-snap-{}", base_name, snapshot_time)
+    }
+
+    /// Create a snapshot
+    pub async fn create_snapshot(
+        self,
+        nvmf_req: *mut spdk_nvmf_request,
+        snapshot_name: &str,
+    ) {
+        extern "C" fn snapshot_done_cb(
+            nvmf_req_ptr: *mut c_void,
+            _lvol_ptr: *mut spdk_lvol,
+            errno: i32,
+        ) {
+            let rsp: &mut spdk_nvme_cpl = unsafe {
+                &mut *spdk_sys::spdk_nvmf_request_get_response(
+                    nvmf_req_ptr as *mut spdk_nvmf_request,
+                )
+            };
+            let nvme_status: &mut spdk_nvme_status =
+                unsafe { &mut rsp.__bindgen_anon_1.status };
```

Comment:
I really *really* do not like this ` __bindgen_anon_1` these are subject to change if we bump things in Cargo.toml. Have you looked into the actual structs (i have not) and see if they are either (a) not whitelisted properly or (b) if they if we can create them "by hand"? 

If we can create them by hand if rather have some small C wrappers (like in logwrapper.c for example) that would return the values we need? 

What do you think? We don't have to do this all at once of course.

Reply:
The problem is the status field is part of an anonymous union with the spdk_nvme_cpl struct so there doesn't seem to be another way to reach it. I figured the struct was stable enough as it's part of the NVMe spec.

Reply:
I've introduced a C helper function to address this.